cmd/main.go

```diff
@@ -49,6 +49,15 @@ func main() {
 		namespacesGroup.DELETE("/:name", routes.DeleteNamespace(client, logger))
 	}
 
+	v1 := engine.Group("/v1")
+	{
+		v1.POST("/secrets", routes.CreateSecret(client, logger))
+		v1.GET("/secrets", routes.GetSecrets(client, logger))
+		v1.GET("/secret", routes.GetSecret(client, logger))
+		v1.PATCH("/secrets", routes.PatchSecret(client, logger))
+		v1.DELETE("/secrets", routes.DeleteSecret(client, logger))
+	}
+
```

Comment:
This will need refactoring after we move to user super-groups, but it's fine for now.

The routes however are not as shown in the design since we decided to move the `namespace` to be a `queryParameter` instead of it being part of the `path` so fix the design docs to reflect the current state.

Reply:
> This will need refactoring after we move to user super-groups, but it's fine for now.

Yes. Let's take care of it properly in a new PR before we'll continue to implementing new API endpoints.

> The routes however are not as shown in the design

Ah yes, it should be modified in the design doc as well. I'll take care of updating design doc together with super-groups in a new PR.

> since we decided to move the namespace to be a queryParameter instead of it being part of the path

I thought we both agreed on implementing only body parameters in secrets API. There is a difference between path parameters, query parameters and body parameters:

- Path parameter is part of the URL, e.g. `/users/:id` where `:id` is a path parameter.
- Query parameter is added in the end of URL, e.g. `/users?name=dana&role=admin`.
- Body parameter is part of the request body, not an URL and is JSON data, e.g. `{ "namespace": "default" }`.

Since I'm expecting body parameters in all secrets API endpoints, I'd deserialize the JSON data using `ShouldBindJSON` and not `ShouldBindQuery` or `ShouldBindUri`.

Reply:
You're correct, I meant body parameters. A wrong choice of words on my end 👍